### ISSUE TITLE
do_learn_shape And do_learn_variation functions

### ISSUE THREAD

**Issue by 7color94** [tntrung/impSDM#9]
Hi, tntrung. Thanks for your implementation.
Actually, I wonder why we need do_learn_shape and do_learn_variation functions.  And after we get ShapeModel and DataVariation, what do we do with it?
Sorry for my silly question.

**Comment by tntrung** [comment]
Hi, as mentioned in the original paper of SDM, the mean and variance of shape, which should be learned from training data, is helpful to generating the initial shapes.

**Comment by 7color94** [comment]
Thanks very much.
And when we calculate delta shape as GroundTruth in training phase, we project delta shape to face bbox to make it in a uniform range [0,1]. But different face images may have different scale, rotation, so do we need transform delta shape to mean shape using similarity transform, reducing the impact of scale and rotation?
Thanks to your answer and help.

**Comment by tntrung** [comment]
Yes, it may be helpful. In my implementation, I just simply scaled the detected face size into the same as the mean shape before doing regression. Alternatively, people can do pre-processing by resizing all training images of detected faces into the same scale, so that the shape normalization can be ignored.

**Comment by 7color94** [comment]
Thanks.
